src/main/java/com/project/stockexchangeappbackend/service/UserServiceImpl.java

```diff
@@ -59,10 +65,27 @@ public User findUserById(Long id) {
     @Transactional(readOnly = true)
     @LogicBusinessMeasureTime
     public User findUserByEmail(String email) {
-        return userRepository.findByEmailIgnoreCase(email)
+        return userRepository.findByEmailIgnoreCase(email.trim())
                 .orElseThrow(() -> new EntityNotFoundException("User Not Found"));
     }
 
+    @Override
+    @LogicBusinessMeasureTime
+    @Transactional
+    public User changeUserPassword(Long id, ChangePasswordDTO changePasswordDTO) {
+        User user = userRepository.findById(id)
+                .orElseThrow(() -> new EntityNotFoundException("User Not Found"));
+        if (!validPassword(changePasswordDTO.getNewPassword())) {
+            throw new AccessDeniedException("New password is too weak.");
+        }
+        if (passwordEncoder.matches(changePasswordDTO.getOldPassword(), user.getPassword())) {
+            user.setPassword(passwordEncoder.encode(changePasswordDTO.getNewPassword()));
+            userRepository.save(user);
+            return user;
+        } else {
+            throw new AccessDeniedException("Passwords do not match.");
```

Comment:
It will throw when when user credential's are incorrect. Not when old and new password don't match

Reply:
Fixed.